ui/src/table.tsx

```diff
@@ -208,7 +217,51 @@ const
     const line = JSON.stringify(row)
     return line.substr(1, line.length - 2)
   }).join('\n'),
-  DataTable = ({ model: m, onFilterChange, items, filteredItems, selection, selectedFilters, isMultiple, isSearchable, groups, sort, reset }: DataTable) => {
+  ContextualMenu = ({ onFilterChange, col, listProps, selectedFiltersRef, setFiltersInBulk }: ContextualMenuState) => {
+    const
+      isFilterChecked = (data: S, key: S) => !!selectedFiltersRef.current && selectedFiltersRef.current[data]?.includes(key),
+      [filters, setFilters] = React.useState(col.cellType?.tag
+        ? Array.from(listProps.items.reduce((_filters, { key, text, data }) => {
+          key.split(',').forEach(key => _filters.set(key, { key, text, data }))
+          return _filters
+        }, new Map<S, Fluent.IContextualMenuItem>()).values())
+        : listProps.items.map(i => ({ ...i, checked: isFilterChecked(i.data, i.key) }))
+      ),
+      selectAll = () => {
+        setFilters(filters.map(i => ({ ...i, checked: true })))
+        setFiltersInBulk(col.key, filters.map(f => f.key))
+      },
+      deselectAll = () => {
+        setFilters(filters.map(i => ({ ...i, checked: false })))
+        setFiltersInBulk(col.key, [])
+      },
+      getOnFilterChangeHandler = (data: S, key: S) => (_ev?: React.FormEvent<HTMLInputElement | HTMLElement>, checked?: boolean) => {
```

Comment:
Odd naming, handlers usually start with on rather than get.

nit
```suggestion
      getOnFilterChangeHandler = (data: S, key: S) => (_ev?: React.FormEvent<HTMLInputElement | HTMLElement>, checked?: B) => {
```

Reply:
Not a handler, it's a high-order function that creates a handler, hence the name

Reply:
Agree, fair point.